lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -13,155 +13,219 @@ namespace PlatformAgnostic
 namespace DateTime
 {
 
-    #define updatePeriod 1000
-
-    const WCHAR *Utility::GetStandardName(size_t *nameLength) {
-        data.UpdateTimeZoneInfo();
-        *nameLength = data.standardZoneNameLength;
-        return data.standardZoneName;
+    static inline bool IsLeap(const int year)
+    {
+        return (0 == (year & 3)) && (0 != (year % 100) || 0 == (year % 400));
     }
 
-    const WCHAR *Utility::GetDaylightName(size_t *nameLength) {
-        // We have an abbreviated standard or daylight name
-        // based on the date and time zone we are in.
-        return GetStandardName(nameLength);
+    static inline int NormalizeYMDYear(const int base_year)
+    {
+        if (base_year < -2100)
+        {
+            return 2100;
+        }
+        else if (base_year < -1900)
+        {
+            return base_year * -1;
+        }
+
+        return base_year;
     }
 
-    void UtilityPlatformData::UpdateTimeZoneInfo()
+    static inline int UpdateToYMDYear(const int base_year, const struct tm *time)
     {
-        uint32 tickCount = GetTickCount();
-        if (tickCount - lastTimeZoneUpdateTickCount > updatePeriod)
+        int year = time->tm_year;
+
+        if (base_year < -2100)
+        {
+            const int diff = year - 2100;
+            year = abs(base_year) - diff;
+        }
+
+        if (base_year < -1900)
         {
-            time_t gtime = time(NULL);
-            struct tm ltime;
+            year *= -1;
+        }
 
-            tzset();
-            localtime_r(&gtime, &ltime);
+        return year;
+    }
 
-            standardZoneNameLength = strlen(ltime.tm_zone);
+    static void YMD_TO_TM(const YMD *ymd, struct tm *time, bool *leap_added)
+    {
+        time->tm_year = NormalizeYMDYear(ymd->year);
```

Comment:
Based on the usage of this method, would YMDYearToTmYear (and the inverse) be better names for these methods?